src/Stache/Query/EntryQueryBuilder.php

```diff
@@ -148,4 +149,40 @@ public function prepareForFakeQuery(): array
 
         return $data;
     }
+
+    public function firstOrCreate(array $attributes = [], array $values = [])
```

Comment:
I know Laravel has [these methods in the query builder](https://github.com/laravel/framework/blob/1b453a84552e8e6bc6368e7038ee4104a54ae01c/src/Illuminate/Database/Eloquent/Builder.php#L565) but I feel like these methods might be better in the `EntryRepository`.

Reply:
Ryan mentioned the same thing in discord.  I understand why putting them in the repository would be better.  I think my main concern would be people expecting to be able to prefilter the query and then call firstOrCreate, etc.

The tests have an example like so:
```php
$entry = Entry::query()
	->where('collection', 'posts')
	->firstOrCreate(
		['id' => 'id-1'],
		['title' => 'Post 1'],
	);
```

Granted, this is a rather simplistic use case, but something more complex might be required?
```php
$entry = Entry::query()
	->when($type, fn ($query) => $query->where('type', $type))
	->firstOrCreate(
		[],
		['title' => 'Product 1'],
	);
```

I _think_ that might be valid when using Laravel's model methods but if the methods are moved to the Statamic repository it might not work?

I can do some testing this evening.

Reply:
>  I think my main concern would be people expecting to be able to prefilter the query and then call firstOrCreate, etc.

Ah okay, I missed that. Let's keep it in the query builder then to keep it inline with how Laravel does it. 👍